Userland/Libraries/LibWasm/AbstractMachine/Operators.h

```diff
@@ -882,51 +882,33 @@ struct VectorFloatUnaryOp {
     }
 };
 
-template<size_t VectorSize, typename Op>
-struct VectorFloatConvertOp {
+template<size_t ResultSize, size_t InputSize, typename ResultType, typename InputType, typename Op>
+struct VectorConvertOp {
     auto operator()(u128 lhs) const
     {
-        using VectorInput = NativeFloatingVectorType<128, VectorSize, NativeFloatingType<128 / VectorSize>>;
-        using VectorResult = NativeVectorType<128 / VectorSize, VectorSize, MakeUnsigned>;
+        using VectorInput = NativeVectorType<128 / InputSize, InputSize, MakeUnsigned>;
+        using VectorResult = NativeVectorType<128 / ResultSize, ResultSize, MakeUnsigned>;
         auto value = bit_cast<VectorInput>(lhs);
         VectorResult result;
         Op op;
-        for (size_t i = 0; i < VectorSize; ++i) {
-            result[i] = op(value[i]);
+        auto size = min(InputSize, ResultSize);
+        for (size_t i = 0; i < size; ++i)
+            result[i] = bit_cast<ResultType>(op(bit_cast<InputType>(value[i])));
+        // FIXME: We shouldn't need this, but the auto-vectorizer sometimes doesn't see that we
```

Comment:
This is fine imo (should keep the fixme though, for when it does get fixed in clang...eventually)